web/src/components/common/Backend.jsx

```diff
@@ -441,20 +441,13 @@ class Backend {
     }
 
     async setUserRoles(mail, firstName, lastName, bio, roles){
-      if (await this.putJson('/api/users', {
+      return this.putJson('/api/users', {
         'email': mail,
         'first_name': firstName,
         'last_name': lastName,
         'bio': bio,
         'roles' : roles
-      })) {
-        if (this.mail == mail) {
```

Comment:
Why did you remove this check? What if I make myself from an admin to a normal user? Every profile update should always trigger a refresh of the profile, if it is mine.